IBM-Cloud/ibm-cloud-cli-sdk
# BREAKING CHANGE: Updated to latest version of go-i18n to fix security vulnerability

# Context

Older versions of go-i18n has been flagged with a security vulnerability so the cli-sdk must be updated to at least v2.0.0 to resolve the vulnerability. There is an option to find another open source project however, go-i18n is the most supported and widely used localization package for Go.


# Breaking Changes
The API between v1.0 and v2.0 for go-i18n is drastically different. There are two points to keep in mind.


- The filename have been updated to match what go-i18n expects when parsing the generating bindata produced by go-bindata. Instead of `en_US.all.json` it has been changed to `all.en_US.json`. Any package that pulls the latest version of the CLI SDK will need to update the translation filenames accordingly. See files in i18n/resources/ for examples.

# Callouts

- The signature method remains the same as shown in the example below.
 ```
T("CloudFoundy Applications  {{.Used}}/{{.Limit}} used",
			map[string]interface{}{
				"Used":  formattedGB(orgUsage.TotalMemoryUsed()),
				"Limit": formattedGB(cmd.cf.CurrentOrganization().QuotaDefinition.InstanceMemoryLimitInMB)}), color.New(color.FgYellow, color.Bold)))

```
- When a `messageID` does not exist in the translation files for a given language other than the default (English). The default message will be used
- When a `messageID` does not exist in any of the translation file. A panic error will be thrown with the error message format: `"<MESSAGE-ID>" not found in language "<language>"`. This panic behavior is provided by go-i18n. If we want to just return an error and not panic I can adjust this accordingly to handle the panic as an error only if the error message is of this type.

# Steps to Test
1. Open a terminal and change directory to the root of the `ibm-cloud-cli-sdk` local repo
2. Run the command `go test ./...`
3. Verify all tests pass
